MekHQ/src/mekhq/gui/adapter/PersonnelTableMouseAdapter.java

```diff
@@ -1413,15 +1413,15 @@ private void maybeShowPopup(MouseEvent e) {
                     } else if (unit.usesSoldiers()) {
                         if (unit.canTakeMoreGunners() && person.canGun(unit.getEntity())) {
                             cbMenuItem = new JCheckBoxMenuItem(unit.getName());
-                            cbMenuItem.setSelected(person.getUnitId() == unit.getId());
+                            cbMenuItem.setSelected(person.getUnitId().equals(unit.getId()));
```

Comment:
Can `person.getUnitId` be null? If so you should swap the order as `unit.getId()` will always be non-null.

Reply:
Yeah, that's a dumb tired programmer mistake.

Reply:
Fixed